kairon/shared/data/processor.py

```diff
@@ -1132,6 +1160,117 @@ def __prepare_training_story_events(self, events, timestamp):
             elif event.type == SlotSet.type_name:
                 yield SlotSet(key=event.name, value=event.value, timestamp=timestamp)
 
+    def __extract_multiflow_story_events_steps(self, step_data, bot, user):
+        step_event = {}
+        component_id = None
+        try:
+            if step_data:
```

Comment:
why db query here to retrieve intent, slot, utterances, actions?
Moreover this is too complex. simplify this